sharktank/sharktank/layers/paged_attention.py

```diff
@@ -126,6 +128,23 @@ def __init__(
         self.cache_dtype = cache_dtype
         self.attn_dtype = attn_dtype
 
+    def unflatten_page_table(
+        self,
+        state: torch.Tensor | SplitPrimitiveTensor | ReplicatedTensor,
+        pipeline: int,
+    ) -> torch.Tensor | SplitPrimitiveTensor | ReplicatedTensor:
+        """Unflattens the 2D page tables to 6D tensors."""
+        shards = [state] if self.shard_count == 1 else state.shards
+        shards = [shard.unflatten(1, self.sub_page_dims[pipeline]) for shard in shards]
+
+        if self.shard_count > 1:
+            return SplitPrimitiveTensor(ts=shards, shard_dim=4, devices=state.devices)
+
+        if pipeline > 1:
+            return ReplicatedTensor(ts=shards, devices=state.devices)
```

Comment:
What if we are doing a mix of pipeline and tensor parallelism? Is that not possible? Can we document that?